File: Builder/HtmlBuilder/WithFluentBuilder/HtmlBuilder.cs
namespace WithFluentBuilder
{
    public class HtmlBuilder
    {
        private readonly string _rootName;
        HtmlElement root = new HtmlElement();

        public HtmlBuilder(string rootName)
        {
            _rootName = rootName;
            root.Tag = rootName;
        }

        public HtmlBuilder AddChild(string tag, string value)
        {
            var e = new HtmlElement(tag, value);
            root.Elements.Add(e);
            return this;
        }

        public override string ToString()
        {
            return root.ToString();
        }

        public void Clear()
        {
            root = new HtmlElement { Tag = _rootName };
        }
    }
}

File: Builder/Python/HtmlBuilder/WithBuilder/html_builder.py
from html_element import HtmlElement


class HtmlBuilder:
    def __init__(self, root_name: str) -> None:
        self.root_name: str = root_name
        self.__root = HtmlElement(root_name)

    def add_child(self, child_name: str, child_text: str) -> None:
        self.__root.elements.append(
            HtmlElement(child_name, child_text)
        )

    def add_child_fluent(self, child_name: str, child_text: str):
        self.__root.elements.append(
            HtmlElement(child_name, child_text)
        )
        return self

    def clear(self) -> None:
        self.__root = HtmlElement(name=self.root_name)

    def __str__(self) -> str:
        return str(self.__root)
